File: py/1051-height-checker.py
class Solution:
    def heightChecker(self, heights: list[int]) -> int:
        freqs = 101 * [0]
        for height in heights:
            freqs[height] += 1

        curr = 1
        ans = 0

        for height in heights:
            while freqs[curr] == 0:
                curr += 1

            if curr != height:
                ans += 1

            freqs[curr] -= 1

        return ans

File: cpp/1051-height-checker.cpp
#include <array>
#include <vector>

class Solution {
    public:
        [[nodiscard]] auto heightChecker(const std::vector<int> &heights)
            -> int {
            auto freqs = std::array<int, 101>{};
            for (const auto height : heights) {
                ++freqs[static_cast<unsigned>(height)];
            }

            auto curr = 1U;
            auto ans = 0;

            for (const auto height : heights) {
                while (freqs[curr] == 0) {
                    ++curr;
                }

                if (curr != static_cast<unsigned>(height)) {
                    ++ans;
                }

                --freqs[curr];
            }

            return ans;
        }
};
